File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/QuestionWithAnswersDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.Category;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class QuestionWithAnswersDTO {
    @NotBlank(message = "{question.text.notblank}")
    private String text;
    @NotNull(message = "{question.category.notnull}")
    private Category category;
    @Size(min = 5, max = 5, message = "{question.answeroptions.size}")
    private List<AnswerOptionFilteredDTO> answerOptions;
}

File: OrienGO-Frontend/src/app/model/dto/QuestionWithAnswers.dto.ts
import { Category } from "../enum/Category.enum";
import { AnswerOptionFilteredDTO } from "./AnswerOptionFiltered.dto";

export interface QuestionWithAnswersDTO {
  text: string;
  category: Category;
  answerOptions: AnswerOptionFilteredDTO[]; // Must always be exactly 5
}